src/Standards/Squiz/Sniffs/Commenting/FunctionCommentSniff.php

```diff
@@ -551,24 +551,45 @@ protected function processParams(File $phpcsFile, $stackPtr, $commentStart)
 
             // Make sure the param name is correct.
             if (isset($realParams[$pos]) === true) {
-                $realName = $realParams[$pos]['name'];
-                if ($realName !== $param['var']) {
+                $realName     = $realParams[$pos]['name'];
+                $paramVarName = $param['var'];
+
+                if ($param['var'][0] === '&') {
+                    // Even when passed by reference, the variable name in $realParams does not have
+                    // a leading '&'. This sniff will accept both '&$var' and '$var' in these cases.
+                    $paramVarName = substr($param['var'], 1);
+
+                    // This makes sure that the 'MissingParamTag' check won't throw a false positive.
+                    $foundParams[(count($foundParams) - 1)] = $paramVarName;
+
+                    if ($realParams[$pos]['pass_by_reference'] !== true) {
```

Comment:
```suggestion
                    if ($realName === $paramVarName && $realParams[$pos]['pass_by_reference'] !== true) {
```

Extra tests to verify why I'm suggesting this change:
```php
/**
 * Test for param tag containing ref, but param in declaration not being by ref.
 *
 * @param string &$foo This should be flagged as ParamNameUnexpectedAmpersandPrefix.
 * @param string &$bar This should be flagged as ParamNameNoMatch.
 * @param string &$baz This should be flagged as ParamNameNoCaseMatch.
 *
 * @return void
 */
function passedByRefMismatch($foo, $bra, $BAZ) {
	return;
}
```

Reply:
This change seems fair. I had considered doing this but decided against it at the time. The reasoning was that if the variable name was incorrect *and* had a rogue leading ampersand, that both faults would be flagged. With the suggested change (which I've committed in 28ed6aec1), users will get one fault identified at a time, so fixing one fault won't make all errors go away, but instead will make the error message change. I can see how having two errors is less desirable, especially in the case where the param tags are out of order and changing the order alone would fix all errors.

Reply:
Good points and I agree that generally speaking errors shouldn't hide behind each other, but in this case, if the variable name doesn't match, the information about whether or not the variable is passed by reference from the function signature is just plain unreliable, which is why I suggested the change.